fedimint-server/src/consensus/server.rs

```diff
@@ -1,106 +1,51 @@
 use fedimint_core::net::peers::PeerConnections;
 use fedimint_core::task::{sleep, RwLock, TaskGroup, TaskHandle};
-use fedimint_core::{NumPeers, PeerId};
-use futures::stream::Peekable;
-use futures::{FutureExt, StreamExt};
-use hbbft::honey_badger::{Batch, HoneyBadger, Message, Step};
-use hbbft::{Epoched, NetworkInfo, Target};
-use itertools::Itertools;
-use jsonrpsee::core::Serialize;
-use rand::rngs::OsRng;
-use rand::{CryptoRng, RngCore};
-use serde::Deserialize;
-use tokio::sync::mpsc;
-use tokio_stream::wrappers::ReceiverStream;
-use tracing::{info, warn};
+use fedimint_core::PeerId;
+use tokio::select;
+use tracing::{debug, info, warn};
 
+use crate::atomic_broadcast::{AtomicBroadcast, Decision, Keychain, Message, Recipient};
 use crate::config::ServerConfig;
-use crate::consensus::{
-    ApiEvent, ConsensusOutcomeConversion, ConsensusProposal, FedimintConsensus,
-    HbbftConsensusOutcome, HbbftSerdeConsensusOutcome,
-};
-use crate::db::{get_global_database_migrations, LastEpochKey, GLOBAL_DATABASE_VERSION};
+use crate::consensus::FedimintConsensus;
+use crate::db::{get_global_database_migrations, AcceptedIndex, GLOBAL_DATABASE_VERSION};
 use crate::fedimint_core::encoding::Encodable;
 use crate::fedimint_core::net::peers::IPeerConnections;
 use crate::net::api::{ConsensusApi, ExpiringCache, InvitationCodesTracker};
 use crate::net::connect::{Connector, TlsTcpConnector};
-use crate::net::peers::{DelayCalculator, PeerConnector, PeerSlice, ReconnectPeerConnections};
+use crate::net::peers::{DelayCalculator, PeerConnector, ReconnectPeerConnections};
 use crate::{LOG_CONSENSUS, LOG_CORE};
-type PeerMessage = (PeerId, EpochMessage);
-
-/// How many epochs ahead of consensus to rejoin
-const NUM_EPOCHS_REJOIN_AHEAD: u64 = 10;
 
 /// How many txs can be stored in memory before blocking the API
-const TRANSACTION_BUFFER_SIZE: usize = 1000;
-
-#[derive(Clone, Debug, Deserialize, Serialize)]
-#[allow(clippy::large_enum_variant)]
-pub enum EpochMessage {
-    Continue(Message<PeerId>),
-    RejoinRequest,
-}
-
-type EpochStep = Step<Vec<SerdeConsensusItem>, PeerId>;
-
-enum EpochTriggerEvent {
-    /// A new event has been sent to us from the API
-    ApiEvent,
-    /// A peer has sent us a consensus message
-    NewMessage(PeerMessage),
-    /// One of our modules triggered an event (e.g. new bitcoin block)
-    ModuleProposalEvent,
-    /// A rejoining peer wants us to run an empty epoch
-    RunEpochRequest,
-}
+const TRANSACTION_BUFFER: usize = 1000;
 
 pub(crate) type LatestContributionByPeer = HashMap<PeerId, u64>;
 
 /// Runs the main server consensus loop
 pub struct ConsensusServer {
     /// Delegate for processing consensus information
-    consensus: FedimintConsensus,
-    /// Receives event notifications from the `ConsensusApi` (triggers epochs)
-    api_receiver: Peekable<ReceiverStream<ApiEvent>>,
-    /// P2P connections for running consensus
-    connections: PeerConnections<EpochMessage>,
+    pub consensus: FedimintConsensus,
```

Comment:
Donno how I feel about making everything `pub` here, but I guess it's OK.

Reply:
Also not final yet